src/node/dht/dht.c

```diff
@@ -191,6 +198,11 @@ result_t add_new_peer(node_ctx_t *ctx, char *peer_host) {
   result_t res_manifest = json_string_to_node_manifest(manifest_str);
   node_manifest_t *manifest = PROPAGATE(res_manifest);
 
+  if (!valid_node_id(manifest)) {
+    free_node_manifest(manifest);
+    return ERR("peer manifest node_id not valid");
```

Comment:
Maybe the error message could include why it is invalid?

Reply:
would `peer manifest node_id invalid due to public key mismatch` work?

Reply:
Yes, that should work.